Userland/Libraries/LibWeb/IndexedDB/Internal/Database.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, stelar7 <dudedbz@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashMap.h>
+#include <LibJS/Heap/GCPtr.h>
+#include <LibWeb/DOM/EventTarget.h>
+#include <LibWeb/IndexedDB/IDBDatabase.h>
+#include <LibWeb/IndexedDB/IDBRequest.h>
+#include <LibWeb/StorageAPI/StorageKey.h>
+
+namespace Web::IndexedDB {
+
+// https://www.w3.org/TR/IndexedDB/#database-construct
+class Database : public Bindings::PlatformObject {
+    WEB_PLATFORM_OBJECT(Database, Bindings::PlatformObject);
+    JS_DECLARE_ALLOCATOR(Database);
+
+public:
+    void set_version(u64 version) { m_version = version; }
+    u64 version() const { return m_version; }
+    String name() const { return m_name; }
+
+    void associate(JS::NonnullGCPtr<IDBDatabase> connection) { m_associated_connections.append(connection); }
+    Vector<JS::NonnullGCPtr<IDBDatabase>> associated_connections() { return m_associated_connections; }
+    Vector<JS::NonnullGCPtr<IDBDatabase>> associated_connections_except(IDBDatabase& connection)
```

Comment:
We should avoid returning a `Vector<NonnullGCPtr<T>>` as it is not safe as the GC is not able to track those pointers.

Probably it should be returning a `MarkedVector<T>`

Reply:
I'm not 100% on how the GC works, but the pointers are still stored in `m_associated_connections`, and will be visited that way?

Reply:
Tim's comment here does a better job explaining it that I can: https://github.com/LadybirdBrowser/ladybird/pull/1854#discussion_r1806807140

Basically, since those GCPtr's are on the heap, and not the stack, the GC is not able to mark them, which can introduce problems if we are passing this vector